src/control/lib/support/log.go

```diff
@@ -47,14 +47,15 @@ const (
 )
 
 type CollectLogSubCmd struct {
-	StopOnError  bool   `short:"s" long:"stop-on-error" description:"Stop the collect-log command on very first error"`
-	TargetFolder string `short:"t" long:"target-folder" description:"Target Folder location where log will be copied"`
-	Archive      bool   `short:"z" long:"archive" description:"Archive the log/config files"`
-	ExtraLogsDir string `short:"c" long:"extra-logs-dir" description:"Collect the Logs from given directory"`
-	LogStartDate string `short:"D" long:"start-date" description:"Specify the start date, the day from log will be collected, Format: MM-DD"`
-	LogEndDate   string `short:"F" long:"end-date" description:"Specify the end date, the day till the log will be collected, Format: MM-DD"`
-	LogStartTime string `short:"S" long:"log-start-time" description:"Specify the log collection start time, Format: HH:MM:SS"`
-	LogEndTime   string `short:"E" long:"log-end-time" description:"Specify the log collection end time, Format: HH:MM:SS"`
+	StopOnError           bool   `short:"s" long:"stop-on-error" description:"Stop the collect-log command on very first error"`
+	TargetFolder          string `short:"t" long:"target-folder" description:"Target Folder location where log will be copied"`
+	Archive               bool   `short:"z" long:"archive" description:"Archive the log/config files"`
+	ExtraLogsDir          string `short:"c" long:"extra-logs-dir" description:"Collect the Logs from given directory"`
+	LogStartDate          string `short:"D" long:"start-date" description:"Specify the start date, the day from log will be collected, Format: MM-DD"`
+	LogEndDate            string `short:"F" long:"end-date" description:"Specify the end date, the day till the log will be collected, Format: MM-DD"`
+	LogStartTime          string `short:"S" long:"log-start-time" description:"Specify the log collection start time, Format: HH:MM:SS"`
+	LogEndTime            string `short:"E" long:"log-end-time" description:"Specify the log collection end time, Format: HH:MM:SS"`
+	ExtraArgsCollectRsync string `short:"C" long:"cloud-storage" description:"Specify the destination in cloud storage"`
```

Comment:
From a UX perspective, it's a little awkward to require that the user both set an environment variable and specify a CLI flag. That's two different configuration paradigms. The [go-flags](https://github.com/jessevdk/go-flags) package does support setting flags via env var, so what might be better here is to add another flag that can be set either on the command line or the environment. For example:
```suggestion
         RsyncAltBin   string `short:"R" long:"rsync-alt-bin" description:"Use an alternate binary in place of rsync for transfers" env:"D_RSYNC_ALT"`
	RsyncAltArgs []string `short:"A" long:"rsync-alt-args" description:"Extra arguments for alternate transfer solution (e.g. cloud storage bucket)" env:"D_RSYNC_ALT_ARGS" env-delim:","`
```

Note that this suggestion also changes the extra arguments into a string slice instead of a single string. If we're building in flexibility to use solutions other than rsync, it would be nice to really make it flexible now.

The other thing to note here is that using `go-flags` for argument parsing keeps the responsibility for that in one place, and makes the configuration explicit. If you want to get fancy, you can do some validation on argument parsing by defining type aliases that implement the [Unmarshaler](https://pkg.go.dev/github.com/jessevdk/go-flags#Unmarshaler) interface. There are examples of this in the other commands.

Using raw environment variables for configuration in other places is harder to reason about and maintain.

Reply:
I had considered that solution. It is simpler. My concern was around security.
This would give the use the ability to do a dmg-execute command ability. T
he alternate binary could be "dd" with the args to overwrite a file. Or it could be to echo a command into a script.

Reply:
> My concern was around security.
This would give the user the ability to do a dmg-execute command ability.

That's a very good point, but isn't that also a concern with the PR as it is? Seems like the best way to address that problem is to allow the user to specify the name of an alternate transfer method and then resolve that name into a known binary in a standard path.

Reply:
Yes that would work. My original plan was to use a destination like "gcs://...." to imply that sort of thing, but that included a bunch of vendor deps.
I was thinking the name could be configurable via the ENV var, but I suppose it could be hard coded to a name like "daos_alt_rsync"